5.5.0: pytest is failing in one unit

I'm trying to package your module as an rpm package. So I'm using the typical PEP517 based build, install and test cycle used on building packages from non-root account.
- `python3 -sBm build -w --no-isolation`
- because I'm calling `build` with `--no-isolation` I'm using during all processes only locally installed modules
- install .whl file in </install/prefix>
- run pytest with PYTHONPATH pointing to sitearch and sitelib inside </install/prefix>

Here is pytest output:
```console
+ PYTHONPATH=/home/tkloczko/rpmbuild/BUILDROOT/python-inflect-5.5.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages:/home/tkloczko/rpmbuild/BUILDROOT/python-inflect-5.5.0-2.fc35.x86_64/usr/lib/python3.8/site-packages
+ /usr/bin/pytest -ra tests
=========================================================================== test session starts ============================================================================
platform linux -- Python 3.8.12, pytest-7.1.1, pluggy-1.0.0
rootdir: /home/tkloczko/rpmbuild/BUILD/inflect-5.5.0, configfile: pytest.ini
plugins: mypy-0.8.1, flake8-1.1.1, enabler-1.2.1, cov-3.0.0, checkdocs-2.7.1, black-0.3.12
collected 111 items

tests/test_an.py ...F.                                                                                                                                               [  4%]
tests/test_classical_all.py ....                                                                                                                                     [  8%]
tests/test_classical_ancient.py ....                                                                                                                                 [ 11%]
tests/test_classical_herd.py ....                                                                                                                                    [ 15%]
tests/test_classical_names.py ....                                                                                                                                   [ 18%]
tests/test_classical_person.py ....                                                                                                                                  [ 22%]
tests/test_classical_zero.py ....                                                                                                                                    [ 26%]
tests/test_compounds.py ............                                                                                                                                 [ 36%]
tests/test_inflections.py ...........                                                                                                                                [ 46%]
tests/test_join.py ....                                                                                                                                              [ 50%]
tests/test_numwords.py .......                                                                                                                                       [ 56%]
tests/test_pl_si.py ....                                                                                                                                             [ 60%]
tests/test_pwd.py ........................................                                                                                                           [ 96%]
tests/test_unicode.py ....                                                                                                                                           [100%]

================================================================================= FAILURES =================================================================================
_______________________________________________________________________________ test session _______________________________________________________________________________
mypy exited with status 1.
============================================================================= warnings summary =============================================================================
../../../../../usr/lib/python3.8/site-packages/_pytest/nodes.py:140: 14 warnings
  /usr/lib/python3.8/site-packages/_pytest/nodes.py:140: PytestRemovedIn8Warning: The (fspath: py.path.local) argument to MypyFile is deprecated. Please use the (path: pathlib.Path) argument instead.
  See https://docs.pytest.org/en/latest/deprecations.html#fspath-argument-for-node-constructors-replaced-with-pathlib-path
    return super().__call__(*k, **kw)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html

---------- coverage: platform linux, python 3.8.12-final-0 -----------
Name                                                                                                                       Stmts   Miss  Cover   Missing
--------------------------------------------------------------------------------------------------------------------------------------------------------
/home/tkloczko/rpmbuild/BUILDROOT/python-inflect-5.5.0-2.fc35.x86_64/usr/lib/python3.8/site-packages/inflect/__init__.py    1190     13    99%   120, 2525, 2530, 2555, 3087, 3093, 3101, 3419, 3506-3508, 3554-3555
tests/test_an.py                                                                                                              18      0   100%
tests/test_classical_all.py                                                                                                   37      4    89%   103-106
tests/test_classical_ancient.py                                                                                               11      0   100%
tests/test_classical_herd.py                                                                                                  11      0   100%
tests/test_classical_names.py                                                                                                 13      0   100%
tests/test_classical_person.py                                                                                                12      0   100%
tests/test_classical_zero.py                                                                                                  11      0   100%
tests/test_compounds.py                                                                                                       32      0   100%
tests/test_inflections.py                                                                                                    188      0   100%
tests/test_join.py                                                                                                            28      0   100%
tests/test_numwords.py                                                                                                        52      0   100%
tests/test_pl_si.py                                                                                                           24      5    79%   22, 33-40
tests/test_pwd.py                                                                                                            495      5    99%   29, 1229-1232
tests/test_unicode.py                                                                                                          8      0   100%
--------------------------------------------------------------------------------------------------------------------------------------------------------
TOTAL                                                                                                                       2130     27    99%

=================================================================================== mypy ===================================================================================
inflect/__init__.py:2019: error: "str" not callable
inflect/__init__.py:2020: error: "List[str]" not callable
inflect/__init__.py:2031: error: Need type annotation for "pl_sb_user_defined" (hint: "pl_sb_user_defined: List[<type>] = ...")
inflect/__init__.py:2032: error: Need type annotation for "pl_v_user_defined" (hint: "pl_v_user_defined: List[<type>] = ...")
inflect/__init__.py:2033: error: Need type annotation for "pl_adj_user_defined" (hint: "pl_adj_user_defined: List[<type>] = ...")
inflect/__init__.py:2034: error: Need type annotation for "si_sb_user_defined" (hint: "si_sb_user_defined: List[<type>] = ...")
inflect/__init__.py:2035: error: Need type annotation for "A_a_user_defined" (hint: "A_a_user_defined: List[<type>] = ...")
inflect/__init__.py:2192: error: Incompatible types in assignment (expression has type "int", variable has type "None")
inflect/__init__.py:3163: error: Unsupported operand types for + ("str" and "Dict[str, str]")
inflect/__init__.py:3163: note: Right operand is of type "Union[str, Dict[str, str]]"
inflect/__init__.py:3168: error: Incompatible return value type (got "Union[str, Dict[str, str]]", expected "Union[str, bool]")
inflect/__init__.py:3173: error: Incompatible return value type (got "Union[str, Dict[str, str]]", expected "Union[str, bool]")
inflect/__init__.py:3598: error: Value of type "None" is not indexable
inflect/__init__.py:3611: error: Value of type "None" is not indexable
inflect/__init__.py:3615: error: Value of type "None" is not indexable
inflect/__init__.py:3622: error: Value of type "None" is not indexable
inflect/__init__.py:3630: error: Value of type "None" is not indexable
inflect/__init__.py:3631: error: Value of type "None" is not indexable
inflect/__init__.py:3638: error: Value of type "None" is not indexable
inflect/__init__.py:3642: error: Value of type "None" is not indexable
inflect/__init__.py:3646: error: Value of type "None" is not indexable
inflect/__init__.py:3648: error: Value of type "None" is not indexable
inflect/__init__.py:3678: error: Value of type "None" is not indexable
inflect/__init__.py:3680: error: Value of type "None" is not indexable
inflect/__init__.py:3722: error: Incompatible types in assignment (expression has type "Dict[str, str]", variable has type "None")
Found 24 errors in 1 file (checked 14 source files)
========================================================================= short test summary info ==========================================================================
FAILED tests/test_an.py::mypy-status
=============================================================== 1 failed, 110 passed, 14 warnings in 48.37s ================================================================

--- a/CHANGES.rst
+++ b/CHANGES.rst
@@ -1,3 +1,8 @@
+v5.5.1
+======
+
+* #150: Rewrite to satisfy type checkers.
+
 v5.5.0
 ======
 

--- a/inflect/__init__.py
+++ b/inflect/__init__.py
@@ -2010,16 +2010,16 @@ DIGIT = re.compile(r"\d")
 
 
 class Words(str):
-    lower: str  # type: ignore
-    split: List[str]  # type: ignore
+    lowered: str
+    split_: List[str]
     first: str
     last: str
 
     def __init__(self, orig) -> None:
-        self.lower = self.lower()
-        self.split = self.split()
-        self.first = self.split[0]
-        self.last = self.split[-1]
+        self.lowered = self.lower()
+        self.split_ = self.split()
+        self.first = self.split_[0]
+        self.last = self.split_[-1]
 
 
 class engine:
@@ -2656,7 +2656,7 @@ class engine:
             return word
 
         for k, v in pl_sb_uninflected_bysize.items():
-            if word.lower[-k:] in v:
+            if word.lowered[-k:] in v:
                 return word
 
         if self.classical_dict["herd"] and word.last.lower() in pl_sb_uninflected_herd:
@@ -2668,7 +2668,7 @@ class engine:
         if mo and mo.group(2) != "":
             return f"{self._plnoun(mo.group(1), 2)}{mo.group(2)}"
 
-        if " a " in word.lower or "-a-" in word.lower:
+        if " a " in word.lowered or "-a-" in word.lowered:
             mo = PL_SB_PREP_DUAL_COMPOUND_RE.search(word)
             if mo and mo.group(2) != "" and mo.group(3) != "":
                 return (
@@ -2677,17 +2677,17 @@ class engine:
                     f"{self._plnoun(mo.group(3))}"
                 )
 
-        if len(word.split) >= 3:
-            for numword in range(1, len(word.split) - 1):
-                if word.split[numword] in pl_prep_list_da:
+        if len(word.split_) >= 3:
+            for numword in range(1, len(word.split_) - 1):
+                if word.split_[numword] in pl_prep_list_da:
                     return " ".join(
-                        word.split[: numword - 1]
-                        + [self._plnoun(word.split[numword - 1], 2)]
-                        + word.split[numword:]
+                        word.split_[: numword - 1]
+                        + [self._plnoun(word.split_[numword - 1], 2)]
+                        + word.split_[numword:]
                     )
 
         # only pluralize denominators in units
-        mo = DENOMINATOR.search(word.lower)
+        mo = DENOMINATOR.search(word.lowered)
         if mo:
             index = len(mo.group("denominator"))
             return f"{self._plnoun(word[:index])}{word[index:]}"
@@ -2696,10 +2696,10 @@ class engine:
         # there is no more than one word following)
         # degree Celsius => degrees Celsius but degree
         # fahrenheit hour => degree fahrenheit hours
-        if len(word.split) >= 2 and word.split[-2] == "degree":
-            return " ".join([self._plnoun(word.first)] + word.split[1:])
+        if len(word.split_) >= 2 and word.split_[-2] == "degree":
+            return " ".join([self._plnoun(word.first)] + word.split_[1:])
 
-        lowered_split = word.lower.split("-")
+        lowered_split = word.lowered.split("-")
         if len(lowered_split) >= 3:
             for numword in range(1, len(lowered_split) - 1):
                 if lowered_split[numword] in pl_prep_list_da:
@@ -2716,20 +2716,23 @@ class engine:
         # HANDLE PRONOUNS
 
         for k, v in pl_pron_acc_keys_bysize.items():
-            if word.lower[-k:] in v:  # ends with accusative pronoun
+            if word.lowered[-k:] in v:  # ends with accusative pronoun
                 for pk, pv in pl_prep_bysize.items():
-                    if word.lower[:pk] in pv:  # starts with a prep
-                        if word.lower.split() == [word.lower[:pk], word.lower[-k:]]:
+                    if word.lowered[:pk] in pv:  # starts with a prep
+                        if word.lowered.split() == [
+                            word.lowered[:pk],
+                            word.lowered[-k:],
+                        ]:
                             # only whitespace in between
-                            return word.lower[:-k] + pl_pron_acc[word.lower[-k:]]
+                            return word.lowered[:-k] + pl_pron_acc[word.lowered[-k:]]
 
         try:
-            return pl_pron_nom[word.lower]
+            return pl_pron_nom[word.lowered]
         except KeyError:
             pass
 
         try:
-            return pl_pron_acc[word.lower]
+            return pl_pron_acc[word.lowered]
         except KeyError:
             pass
 
@@ -2753,10 +2756,10 @@ class engine:
                 f"{pl_sb_irregular_compound[(' '.join(lowered_split[-2:])).lower()]}"
             )
 
-        if word.lower[-3:] == "quy":
+        if word.lowered[-3:] == "quy":
             return f"{word[:-1]}ies"
 
-        if word.lower[-6:] == "person":
+        if word.lowered[-6:] == "person":
             if self.classical_dict["persons"]:
                 return f"{word}s"
             else:
@@ -2764,40 +2767,40 @@ class engine:
 
         # HANDLE FAMILIES OF IRREGULAR PLURALS
 
-        if word.lower[-3:] == "man":
+        if word.lowered[-3:] == "man":
             for k, v in pl_sb_U_man_mans_bysize.items():
-                if word.lower[-k:] in v:
+                if word.lowered[-k:] in v:
                     return f"{word}s"
             for k, v in pl_sb_U_man_mans_caps_bysize.items():
                 if word[-k:] in v:
                     return f"{word}s"
             return f"{word[:-3]}men"
-        if word.lower[-5:] == "mouse":
+        if word.lowered[-5:] == "mouse":
             return f"{word[:-5]}mice"
-        if word.lower[-5:] == "louse":
+        if word.lowered[-5:] == "louse":
             v = pl_sb_U_louse_lice_bysize.get(len(word))
-            if v and word.lower in v:
+            if v and word.lowered in v:
                 return f"{word[:-5]}lice"
             return f"{word}s"
-        if word.lower[-5:] == "goose":
+        if word.lowered[-5:] == "goose":
             return f"{word[:-5]}geese"
-        if word.lower[-5:] == "tooth":
+        if word.lowered[-5:] == "tooth":
             return f"{word[:-5]}teeth"
-        if word.lower[-4:] == "foot":
+        if word.lowered[-4:] == "foot":
             return f"{word[:-4]}feet"
-        if word.lower[-4:] == "taco":
+        if word.lowered[-4:] == "taco":
             return f"{word[:-5]}tacos"
 
-        if word.lower == "die":
+        if word.lowered == "die":
             return "dice"
 
         # HANDLE UNASSIMILATED IMPORTS
 
-        if word.lower[-4:] == "ceps":
+        if word.lowered[-4:] == "ceps":
             return word
-        if word.lower[-4:] == "zoon":
+        if word.lowered[-4:] == "zoon":
             return f"{word[:-2]}a"
-        if word.lower[-3:] in ("cis", "sis", "xis"):
+        if word.lowered[-3:] in ("cis", "sis", "xis"):
             return f"{word[:-2]}es"
 
         for lastlet, d, numend, post in (
@@ -2809,19 +2812,19 @@ class engine:
             ("n", pl_sb_U_on_a_bysize, -2, "a"),
             ("a", pl_sb_U_a_ae_bysize, None, "e"),
         ):
-            if word.lower[-1] == lastlet:  # this test to add speed
+            if word.lowered[-1] == lastlet:  # this test to add speed
                 for k, v in d.items():
-                    if word.lower[-k:] in v:
+                    if word.lowered[-k:] in v:
                         return word[:numend] + post
 
         # HANDLE INCOMPLETELY ASSIMILATED IMPORTS
 
         if self.classical_dict["ancient"]:
-            if word.lower[-4:] == "trix":
+            if word.lowered[-4:] == "trix":
                 return f"{word[:-1]}ces"
-            if word.lower[-3:] in ("eau", "ieu"):
+            if word.lowered[-3:] in ("eau", "ieu"):
                 return f"{word}x"
-            if word.lower[-3:] in ("ynx", "inx", "anx") and len(word) > 4:
+            if word.lowered[-3:] in ("ynx", "inx", "anx") and len(word) > 4:
                 return f"{word[:-1]}ges"
 
             for lastlet, d, numend, post in (
@@ -2837,9 +2840,9 @@ class engine:
                 ("o", pl_sb_C_o_i_bysize, -1, "i"),
                 ("n", pl_sb_C_on_a_bysize, -2, "a"),
             ):
-                if word.lower[-1] == lastlet:  # this test to add speed
+                if word.lowered[-1] == lastlet:  # this test to add speed
                     for k, v in d.items():
-                        if word.lower[-k:] in v:
+                        if word.lowered[-k:] in v:
                             return word[:numend] + post
 
             for d, numend, post in (
@@ -2847,7 +2850,7 @@ class engine:
                 (pl_sb_C_im_bysize, None, "im"),
             ):
                 for k, v in d.items():
-                    if word.lower[-k:] in v:
+                    if word.lowered[-k:] in v:
                         return word[:numend] + post
 
         # HANDLE SINGULAR NOUNS ENDING IN ...s OR OTHER SILIBANTS
@@ -2856,47 +2859,47 @@ class engine:
             return f"{word}es"
 
         for k, v in pl_sb_singular_s_bysize.items():
-            if word.lower[-k:] in v:
+            if word.lowered[-k:] in v:
                 return f"{word}es"
 
-        if word.lower[-2:] == "es" and word[0] == word[0].upper():
+        if word.lowered[-2:] == "es" and word[0] == word[0].upper():
             return f"{word}es"
 
-        if word.lower[-1] == "z":
+        if word.lowered[-1] == "z":
             for k, v in pl_sb_z_zes_bysize.items():
-                if word.lower[-k:] in v:
+                if word.lowered[-k:] in v:
                     return f"{word}es"
 
-            if word.lower[-2:-1] != "z":
+            if word.lowered[-2:-1] != "z":
                 return f"{word}zes"
 
-        if word.lower[-2:] == "ze":
+        if word.lowered[-2:] == "ze":
             for k, v in pl_sb_ze_zes_bysize.items():
-                if word.lower[-k:] in v:
+                if word.lowered[-k:] in v:
                     return f"{word}s"
 
-        if word.lower[-2:] in ("ch", "sh", "zz", "ss") or word.lower[-1] == "x":
+        if word.lowered[-2:] in ("ch", "sh", "zz", "ss") or word.lowered[-1] == "x":
             return f"{word}es"
 
         # HANDLE ...f -> ...ves
 
-        if word.lower[-3:] in ("elf", "alf", "olf"):
+        if word.lowered[-3:] in ("elf", "alf", "olf"):
             return f"{word[:-1]}ves"
-        if word.lower[-3:] == "eaf" and word.lower[-4:-3] != "d":
+        if word.lowered[-3:] == "eaf" and word.lowered[-4:-3] != "d":
             return f"{word[:-1]}ves"
-        if word.lower[-4:] in ("nife", "life", "wife"):
+        if word.lowered[-4:] in ("nife", "life", "wife"):
             return f"{word[:-2]}ves"
-        if word.lower[-3:] == "arf":
+        if word.lowered[-3:] == "arf":
             return f"{word[:-1]}ves"
 
         # HANDLE ...y
 
-        if word.lower[-1] == "y":
-            if word.lower[-2:-1] in "aeiou" or len(word) == 1:
+        if word.lowered[-1] == "y":
+            if word.lowered[-2:-1] in "aeiou" or len(word) == 1:
                 return f"{word}s"
 
             if self.classical_dict["names"]:
-                if word.lower[-1] == "y" and word[0] == word[0].upper():
+                if word.lowered[-1] == "y" and word[0] == word[0].upper():
                     return f"{word}s"
 
             return f"{word[:-1]}ies"
@@ -2907,13 +2910,13 @@ class engine:
             return f"{word}s"
 
         for k, v in pl_sb_U_o_os_bysize.items():
-            if word.lower[-k:] in v:
+            if word.lowered[-k:] in v:
                 return f"{word}s"
 
-        if word.lower[-2:] in ("ao", "eo", "io", "oo", "uo"):
+        if word.lowered[-2:] in ("ao", "eo", "io", "oo", "uo"):
             return f"{word}s"
 
-        if word.lower[-1] == "o":
+        if word.lowered[-1] == "o":
             return f"{word}es"
 
         # OTHERWISE JUST ADD ...s
@@ -2970,29 +2973,29 @@ class engine:
         if WHITESPACE.search(word):
             return False
 
-        if words.lower == "quizzes":
+        if words.lowered == "quizzes":
             return "quiz"
 
         # HANDLE STANDARD 3RD PERSON (CHOP THE ...(e)s OFF SINGLE WORDS)
 
         if (
-            words.lower[-4:] in ("ches", "shes", "zzes", "sses")
-            or words.lower[-3:] == "xes"
+            words.lowered[-4:] in ("ches", "shes", "zzes", "sses")
+            or words.lowered[-3:] == "xes"
         ):
             return words[:-2]
 
-        if words.lower[-3:] == "ies" and len(words) > 3:
-            return words.lower[:-3] + "y"
+        if words.lowered[-3:] == "ies" and len(words) > 3:
+            return words.lowered[:-3] + "y"
 
         if (
             words.last.lower() in pl_v_oes_oe
-            or words.lower[-4:] in pl_v_oes_oe_endings_size4
-            or words.lower[-5:] in pl_v_oes_oe_endings_size5
+            or words.lowered[-4:] in pl_v_oes_oe_endings_size4
+            or words.lowered[-5:] in pl_v_oes_oe_endings_size5
         ):
             return words[:-1]
 
-        if words.lower.endswith("oes") and len(words) > 3:
-            return words.lower[:-2]
+        if words.lowered.endswith("oes") and len(words) > 3:
+            return words.lowered[:-2]
 
         mo = ENDS_WITH_S.search(words)
         if mo:
@@ -3109,7 +3112,7 @@ class engine:
             return word
 
         for k, v in pl_sb_uninflected_bysize.items():
-            if words.lower[-k:] in v:
+            if words.lowered[-k:] in v:
                 return word
 
         if self.classical_dict["herd"] and words.last.lower() in pl_sb_uninflected_herd:
@@ -3124,7 +3127,7 @@ class engine:
         if mo and mo.group(2) != "":
             return f"{self._sinoun(mo.group(1), 1, gender=gender)}{mo.group(2)}"
 
-        space_split = words.lower.split(" ")
+        space_split = words.lowered.split(" ")
         if len(space_split) >= 3:
             for numword in range(1, len(space_split) - 1):
                 if space_split[numword] in pl_prep_list_da:
@@ -3139,7 +3142,7 @@ class engine:
                         space_split[: numword - 1] + sinoun_box + space_split[numword:]
                     )
 
-        dash_split = words.lower.split("-")
+        dash_split = words.lowered.split("-")
         if len(dash_split) >= 3:
             for numword in range(1, len(dash_split) - 1):
                 if dash_split[numword] in pl_prep_list_da:
@@ -3155,22 +3158,25 @@ class engine:
         # HANDLE PRONOUNS
 
         for k, v in si_pron_acc_keys_bysize.items():
-            if words.lower[-k:] in v:  # ends with accusative pronoun
+            if words.lowered[-k:] in v:  # ends with accusative pronoun
                 for pk, pv in pl_prep_bysize.items():
-                    if words.lower[:pk] in pv:  # starts with a prep
-                        if words.lower.split() == [words.lower[:pk], words.lower[-k:]]:
+                    if words.lowered[:pk] in pv:  # starts with a prep
+                        if words.lowered.split() == [
+                            words.lowered[:pk],
+                            words.lowered[-k:],
+                        ]:
                             # only whitespace in between
-                            return words.lower[:-k] + get_si_pron(
-                                "acc", words.lower[-k:], gender
+                            return words.lowered[:-k] + get_si_pron(
+                                "acc", words.lowered[-k:], gender
                             )
 
         try:
-            return get_si_pron("nom", words.lower, gender)
+            return get_si_pron("nom", words.lowered, gender)
         except KeyError:
             pass
 
         try:
-            return get_si_pron("acc", words.lower, gender)
+            return get_si_pron("acc", words.lowered, gender)
         except KeyError:
             pass
 
@@ -3193,46 +3199,46 @@ class engine:
                 si_sb_irregular_compound[(" ".join(dash_split[-2:])).lower()],
             )
 
-        if words.lower[-5:] == "quies":
+        if words.lowered[-5:] == "quies":
             return word[:-3] + "y"
 
-        if words.lower[-7:] == "persons":
+        if words.lowered[-7:] == "persons":
             return word[:-1]
-        if words.lower[-6:] == "people":
+        if words.lowered[-6:] == "people":
             return word[:-4] + "rson"
 
         # HANDLE FAMILIES OF IRREGULAR PLURALS
 
-        if words.lower[-4:] == "mans":
+        if words.lowered[-4:] == "mans":
             for k, v in si_sb_U_man_mans_bysize.items():
-                if words.lower[-k:] in v:
+                if words.lowered[-k:] in v:
                     return word[:-1]
             for k, v in si_sb_U_man_mans_caps_bysize.items():
                 if word[-k:] in v:
                     return word[:-1]
-        if words.lower[-3:] == "men":
+        if words.lowered[-3:] == "men":
             return word[:-3] + "man"
-        if words.lower[-4:] == "mice":
+        if words.lowered[-4:] == "mice":
             return word[:-4] + "mouse"
-        if words.lower[-4:] == "lice":
+        if words.lowered[-4:] == "lice":
             v = si_sb_U_louse_lice_bysize.get(len(word))
-            if v and words.lower in v:
+            if v and words.lowered in v:
                 return word[:-4] + "louse"
-        if words.lower[-5:] == "geese":
+        if words.lowered[-5:] == "geese":
             return word[:-5] + "goose"
-        if words.lower[-5:] == "teeth":
+        if words.lowered[-5:] == "teeth":
             return word[:-5] + "tooth"
-        if words.lower[-4:] == "feet":
+        if words.lowered[-4:] == "feet":
             return word[:-4] + "foot"
 
-        if words.lower == "dice":
+        if words.lowered == "dice":
             return "die"
 
         # HANDLE UNASSIMILATED IMPORTS
 
-        if words.lower[-4:] == "ceps":
+        if words.lowered[-4:] == "ceps":
             return word
-        if words.lower[-3:] == "zoa":
+        if words.lowered[-3:] == "zoa":
             return word[:-1] + "on"
 
         for lastlet, d, unass_numend, post in (
@@ -3244,20 +3250,20 @@ class engine:
             ("a", si_sb_U_on_a_bysize, -1, "on"),
             ("e", si_sb_U_a_ae_bysize, -1, ""),
         ):
-            if words.lower[-1] == lastlet:  # this test to add speed
+            if words.lowered[-1] == lastlet:  # this test to add speed
                 for k, v in d.items():
-                    if words.lower[-k:] in v:
+                    if words.lowered[-k:] in v:
                         return word[:unass_numend] + post
 
         # HANDLE INCOMPLETELY ASSIMILATED IMPORTS
 
         if self.classical_dict["ancient"]:
 
-            if words.lower[-6:] == "trices":
+            if words.lowered[-6:] == "trices":
                 return word[:-3] + "x"
-            if words.lower[-4:] in ("eaux", "ieux"):
+            if words.lowered[-4:] in ("eaux", "ieux"):
                 return word[:-1]
-            if words.lower[-5:] in ("ynges", "inges", "anges") and len(word) > 6:
+            if words.lowered[-5:] in ("ynges", "inges", "anges") and len(word) > 6:
                 return word[:-3] + "x"
 
             for lastlet, d, class_numend, post in (
@@ -3275,15 +3281,15 @@ class engine:
                 ("m", si_sb_C_im_bysize, -2, ""),
                 ("i", si_sb_C_i_bysize, -1, ""),
             ):
-                if words.lower[-1] == lastlet:  # this test to add speed
+                if words.lowered[-1] == lastlet:  # this test to add speed
                     for k, v in d.items():
-                        if words.lower[-k:] in v:
+                        if words.lowered[-k:] in v:
                             return word[:class_numend] + post
 
         # HANDLE PLURLS ENDING IN uses -> use
 
         if (
-            words.lower[-6:] == "houses"
+            words.lowered[-6:] == "houses"
             or word in si_sb_uses_use_case
             or words.last.lower() in si_sb_uses_use
         ):
@@ -3297,7 +3303,7 @@ class engine:
         # HANDLE PLURLS ENDING IN oes -> oe
 
         if (
-            words.lower[-5:] == "shoes"
+            words.lowered[-5:] == "shoes"
             or word in si_sb_oes_oe_case
             or words.last.lower() in si_sb_oes_oe
         ):
@@ -3312,10 +3318,10 @@ class engine:
             return word[:-2]
 
         for k, v in si_sb_singular_s_bysize.items():
-            if words.lower[-k:] in v:
+            if words.lowered[-k:] in v:
                 return word[:-2]
 
-        if words.lower[-4:] == "eses" and word[0] == word[0].upper():
+        if words.lowered[-4:] == "eses" and word[0] == word[0].upper():
             return word[:-2]
 
         if words.last.lower() in si_sb_z_zes:
@@ -3324,19 +3330,19 @@ class engine:
         if words.last.lower() in si_sb_zzes_zz:
             return word[:-2]
 
-        if words.lower[-4:] == "zzes":
+        if words.lowered[-4:] == "zzes":
             return word[:-3]
 
         if word in si_sb_ches_che_case or words.last.lower() in si_sb_ches_che:
             return word[:-1]
 
-        if words.lower[-4:] in ("ches", "shes"):
+        if words.lowered[-4:] in ("ches", "shes"):
             return word[:-2]
 
         if words.last.lower() in si_sb_xes_xe:
             return word[:-1]
 
-        if words.lower[-3:] == "xes":
+        if words.lowered[-3:] == "xes":
             return word[:-2]
 
         # HANDLE ...f -> ...ves
@@ -3344,44 +3350,44 @@ class engine:
         if word in si_sb_ves_ve_case or words.last.lower() in si_sb_ves_ve:
             return word[:-1]
 
-        if words.lower[-3:] == "ves":
-            if words.lower[-5:-3] in ("el", "al", "ol"):
+        if words.lowered[-3:] == "ves":
+            if words.lowered[-5:-3] in ("el", "al", "ol"):
                 return word[:-3] + "f"
-            if words.lower[-5:-3] == "ea" and word[-6:-5] != "d":
+            if words.lowered[-5:-3] == "ea" and word[-6:-5] != "d":
                 return word[:-3] + "f"
-            if words.lower[-5:-3] in ("ni", "li", "wi"):
+            if words.lowered[-5:-3] in ("ni", "li", "wi"):
                 return word[:-3] + "fe"
-            if words.lower[-5:-3] == "ar":
+            if words.lowered[-5:-3] == "ar":
                 return word[:-3] + "f"
 
         # HANDLE ...y
 
-        if words.lower[-2:] == "ys":
-            if len(words.lower) > 2 and words.lower[-3] in "aeiou":
+        if words.lowered[-2:] == "ys":
+            if len(words.lowered) > 2 and words.lowered[-3] in "aeiou":
                 return word[:-1]
 
             if self.classical_dict["names"]:
-                if words.lower[-2:] == "ys" and word[0] == word[0].upper():
+                if words.lowered[-2:] == "ys" and word[0] == word[0].upper():
                     return word[:-1]
 
-        if words.lower[-3:] == "ies":
+        if words.lowered[-3:] == "ies":
             return word[:-3] + "y"
 
         # HANDLE ...o
 
-        if words.lower[-2:] == "os":
+        if words.lowered[-2:] == "os":
 
             if words.last.lower() in si_sb_U_o_os_complete:
                 return word[:-1]
 
             for k, v in si_sb_U_o_os_bysize.items():
-                if words.lower[-k:] in v:
+                if words.lowered[-k:] in v:
                     return word[:-1]
 
-            if words.lower[-3:] in ("aos", "eos", "ios", "oos", "uos"):
+            if words.lowered[-3:] in ("aos", "eos", "ios", "oos", "uos"):
                 return word[:-1]
 
-        if words.lower[-3:] == "oes":
+        if words.lowered[-3:] == "oes":
             return word[:-2]
 
         # UNASSIMILATED IMPORTS FINAL RULE
@@ -3391,7 +3397,7 @@ class engine:
 
         # OTHERWISE JUST REMOVE ...s
 
-        if words.lower[-1] == "s":
+        if words.lowered[-1] == "s":
             return word[:-1]
 
         # COULD NOT FIND SINGULAR

--- a/inflect/__init__.py
+++ b/inflect/__init__.py
@@ -1689,7 +1689,7 @@ def get_si_pron(thecase, word, gender) -> str:
     except KeyError:
         raise  # not a pronoun
     try:
-        return cast(str, sing[gender])  # has several types due to gender
+        return sing[gender]  # has several types due to gender
     except TypeError:
         return cast(str, sing)  # answer independent of gender
 

--- a/inflect/__init__.py
+++ b/inflect/__init__.py
@@ -2026,7 +2026,7 @@ class engine:
     def __init__(self) -> None:
 
         self.classical_dict = def_classical.copy()
-        self.persistent_count = None
+        self.persistent_count: Optional[int] = None
         self.mill_count = 0
         self.pl_sb_user_defined: List[str] = []
         self.pl_v_user_defined: List[str] = []

--- a/inflect/__init__.py
+++ b/inflect/__init__.py
@@ -61,6 +61,7 @@ from typing import (
     Tuple,
     Callable,
     Sequence,
+    cast,
 )
 
 
@@ -1682,15 +1683,15 @@ si_pron_acc_keys = enclose("|".join(si_pron["acc"]))
 si_pron_acc_keys_bysize = bysize(si_pron["acc"])
 
 
-def get_si_pron(thecase, word, gender) -> Union[str, Dict[str, str]]:
+def get_si_pron(thecase, word, gender) -> str:
     try:
         sing = si_pron[thecase][word]
     except KeyError:
         raise  # not a pronoun
     try:
-        return sing[gender]  # has several types due to gender
+        return cast(str, sing[gender])  # has several types due to gender
     except TypeError:
-        return sing  # answer independent of gender
+        return cast(str, sing)  # answer independent of gender
 
 
 # These dictionaries group verbs by first, second and third person

--- a/inflect/__init__.py
+++ b/inflect/__init__.py
@@ -2028,11 +2028,11 @@ class engine:
         self.classical_dict = def_classical.copy()
         self.persistent_count = None
         self.mill_count = 0
-        self.pl_sb_user_defined = []
-        self.pl_v_user_defined = []
-        self.pl_adj_user_defined = []
-        self.si_sb_user_defined = []
-        self.A_a_user_defined = []
+        self.pl_sb_user_defined: List[str] = []
+        self.pl_v_user_defined: List[str] = []
+        self.pl_adj_user_defined: List[str] = []
+        self.si_sb_user_defined: List[str] = []
+        self.A_a_user_defined: List[str] = []
         self.thegender = "neuter"
         self._number_args = None
 

--- a/inflect/__init__.py
+++ b/inflect/__init__.py
@@ -2035,7 +2035,15 @@ class engine:
         self.si_sb_user_defined: List[str] = []
         self.A_a_user_defined: List[str] = []
         self.thegender = "neuter"
-        self._number_args = None
+        self.__number_args: Optional[Dict[str, str]] = None
+
+    @property
+    def _number_args(self):
+        return cast(Dict[str, str], self.__number_args)
+
+    @_number_args.setter
+    def _number_args(self, val):
+        self.__number_args = val
 
     deprecated_methods = dict(
         pl="plural",

